src/librustc/traits/select.rs

```diff
@@ -78,8 +77,40 @@ pub struct SelectionContext<'cx, 'tcx:'cx> {
     /// there is no type that the user could *actually name* that
     /// would satisfy it. This avoids crippling inference, basically.
     intercrate: bool,
+
+    /// Obligations generated during inference. If None, no obligations are to be collected (e.g.
+    /// this is swapped out to a `None` when probing).
+    inferred_obligations: Option<PredicateObligations<'tcx>>
```

Comment:
A'ight, and do you have a preference between this PR and #33451? The basic gist of the difference is that this one has a `SelectionOk` ok-result while the other one puts the inferred obligations into the `nested` member of each `Vtable*` variant.